### ISSUE TITLE
Can we replace bigmemory with a small ALTREP?

### ISSUE THREAD

**Issue by kaneplusplus** [kaneplusplus/bigmemory#74]

**Comment by cdeterman** [comment]
Not sure, certainly an interesting question.  The 'example' package, `simplemmap`, demonstrating this can be found [here](https://github.com/ltierney/Rpkg-simplemmap).

**Comment by privefl** [comment]
I found [this presentation](https://www.r-project.org/dsc/2017/slides/dsc2017.pdf).

Maybe @ltierney could give some insight.

**Comment by kaneplusplus** [comment]
@cdeterman and @privefl after looking at the slides I think we can do this and I like the idea of making a `big.matrix` a first-class R object. I guess the question now is should we look closer at doing this? If we should, what will this look like? Will it be better than the current package or are we essentially reimplementing what we currently have?

**Comment by cdeterman** [comment]
I don't think it is a re-implementation as the 'shared memory' aspect of `bigmemory` is rather unique.  I think this ALTREP representation is just finding a way to make the objects more compact.

**Comment by kaneplusplus** [comment]
@cdeterman It has been marketed for compact object representation but overriding `duplicate` could be used so that copied `big.matrix` objects can have copy-on-write behavior. If you are not going to make too many changes to a mmaped object, you could even cause a copied object to trigger a private mapping.

The [vignette](https://svn.r-project.org/R/branches/ALTREP/ALTREP.html) is also a good resource.
